reactive.js

```diff
@@ -0,0 +1,56 @@
+/* reactive engine. the code here combines functions exposed by the
+ * library into a single function to be run. i keep all code depending
+ * on Bacon here, so if somebody will want to stop using Bacon in the
+ * future, it will be easier */
+var lib = require('./lib2');
+var Bacon = require('bacon');
+
+// this is an adapter function. it gets a Pouch changes object and it
+// returns a Bacon stream
+function changesToStream(changes) {
+  return Bacon.fromBinder(function(sink) {
+    changes
+      .on('change', sink)
+      .on('error', function(text) {
+        sink(new Bacon.Error(text));
+      })
+      .on('complete', function() {
+        sink(new Backn.End());
+      });
+  });
+}
+
+module.expose = function(options) {
+  /* the initial configuration document is fetched from the
+   * database. this is updated when a change is detected */
+  var withOptions = lib.withOptions(options),
```

Comment:
yeah i know. the rationale for this is that everything in `var` is declarative. so here we have a big block all made of variables depending on each other, and we just need a single statement, at the end of the file